dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -871,11 +872,24 @@ public String toString() {
   public String getDefinitionOrDefault(TeradataLogsConnectorProperty property) {
     String stringValue = getDefinition(property);
     if (StringUtils.isEmpty(stringValue)) {
-      return property.getDefaultValue();
+      Optional<String> defaultValue = property.getDefaultValue();
+      Preconditions.checkState(
+          defaultValue.isPresent(),
+          "The default value is not defined for property='%s'.",
+          property);
+      return defaultValue.get();
     }
     return stringValue;
   }
 
+  public Optional<String> getDefinitionOrDefaultIfDefined(TeradataLogsConnectorProperty property) {
+    String stringValue = getDefinition(property);
+    if (StringUtils.isEmpty(stringValue)) {
+      return property.getDefaultValue();
+    }
+    return Optional.of(stringValue);
+  }
```

Comment:
For a quick solution, sticking with the local convention of using nullable String would make this easier to merge. Particularly if this is meant to be a temporary solution, making the API change to optional may be introducing too much risk.

Reply:
I've changed the Optional-related code to `null` handling.